### ISSUE TITLE
OLED board version 2 does not display more than 1 row at a time

### ISSUE THREAD

**Issue by mppowell** [mppowell/midi-controller#2]
After redesigning the OLED board to resolve previous issue, I again found that connecting the second row of OLEDs resulted in none of the OLEDs working properly. They would produce a strange, corrupted text, and nothing showed up on the IPS display (which led me ultimately to the answer). When I would turn the page selection encoder, the text would shift by a regular amount each time I would flip it back. This no longer seemed like SPI corruption but rather that the cursor or display size was getting mishandled.
![IMG_3455](https://github.com/user-attachments/assets/be088751-d004-4d9b-b75e-d1b0b5bb0325)
![IMG_3454](https://github.com/user-attachments/assets/41119439-c906-4bba-92c9-647ec350731b)

**Comment by mppowell** [comment]
When I commented out all code related to the IPS display, I discovered that all the OLEDs actually worked! This eliminated the idea that the issue was related to SPI corruption and it eliminated concerns over there not being sufficient power to turn on the screens. When I started to add IPS functionality back in, the OLEDs broke again almost immediately. Even just initializing the IPS display caused the corrupted view to return.

Interestingly, during this initialization, the screen resolution is configured. And the IPS screen resolution is different from that of the OLED screens. Both libraries, call the Adafruit_GFX library, so I wondered if perhaps a global variable related to the screen resolution was being overwritten that caused the cursor to end up in the wrong position for the OLEDs each time they got overwritten.

**Comment by mppowell** [comment]
I discovered the bug which took 2 critical steps. First, when initializing the IPS display, I needed to put it into "SPI_MODE2" (by default it is set to SPI_MODE0). However, I can't find anywhere in the documentation for this display that it uses SPI_MODE2, nor did it have any issue in SPI_MODE0 when used alone, or even with one row of the OLEDs. So I don't fully understand why this worked.

This worked, but resulted in the box showing the selected page on the menu, not to be reset when changing the page. To resolve this issue, I had to re-call the set-rotation function each time I updated the IPS display. I assume this puts the cursor in the correct spot before making any changes.

The issue is resolved, but I am not entirely sure why...
